%\VignetteIndexEntry{Using PASTIS}
\documentclass[a4paper]{article}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{caption}
\captionsetup{belowskip=12pt,aboveskip=4pt}

\title{PASTIS: Phylogenetic Assembly with Soft Taxonomic InferenceS}
\author{Klaas Hartmann, Gavin Thomas, Aki Mimoto, Arne Mooers, 
\\Jeffrey Joy, Walter Jetz}

\begin{document}

\maketitle

\begin{center}




\textit{
\\A bright motmot was acting quite rowdy-
\\weaving and squawking quite loudly
\\"Pastis is delise"
\\he burped with a sneeze
\\"but it seems to make everything cloudy"
\\  --Arne Mooers}
\end{center}

\newpage


\tableofcontents
\addtocontents {toc}{~\hfill\textbf{Page}\par}
\section{Introduction}

PASTIS provides a method for producing fully resolved phylogenetic tree distributions by combining sequence data, taxonomic information and tree constraints. The position of some taxa (particularly those with limited or no sequence information) may be highly uncertain. The PASTIS philosophy is to reflect this uncertainty in the position of these taxa rather than simply omitting these taxa or representing them as unresolved polytomies. For taxa without sequence data placement in the tree is achieved using available taxonomic information (e.g. clade membership). Entire clades may also be included in this manner.

PASTIS assimilates the available information for an analysis into a single mrBayes input file. Beyond the features usually found in a mrBayes input file, an analysis created by PASTIS contains a (potentially) extensive set of generated constraints based on taxonomic information and the provided constraint tree. 

After generating the mrBayes input file with PASTIS, the user runs mrBayes (>=3.2) as per usual (this permits use of the standard high performance computing interfaces). After mrBayes finishes execution the user can analyse the generated distribution of trees using their preferred method. PASTIS also provides a method for verifying that taxa have been placed appropriately in the tree.

Throughout PASTIS and this guide the following terminology is used:\\

\textit{Patch:} The part of the tree of life being investigated.

\textit{Clade:} Groups within the patch are referred to as clades. This analysis assumes that clade membership for all taxa is specified.

\textit{Taxa:} The organisational unit corresponding to the leaves, usually species, or sub-species.\\

This document provides an overview of the mechanism for using PASTIS. It \textbf{does not} provide a motivation or description of the algorithms used by PASTIS. Readers are referred to the paper for an in depth discussion of the algorithms and choices.


\section{Quick overview}
This section provides a very brief overview and may be most suitable for readers comfortable with R and mrBayes. Subsequent sections provide a more detailed description.

The main function of interest to users is pastis\_simple. If the correct files are in your working directory in R, and PASTIS is installed, then running PASTIS can be as simple as:
\\
\\library(pastis)
\\pastis\_simple('PatchName')\\

Where PatchName is the name for your analysis. This will generate a file called PatchName.nexus which you would then use as input for mrBayes (version >=3.2).

To get started quickly example input files are available \href{http://dx.doi.org/10.6084/m9.figshare.692180}{here}. These provide a good reference point for the range of possible PASTIS inputs. Files beginning pastis\_data are simple examples of the most basic PASTIS analyses using only an incomplete phylogeny and a species list. The files beginning Accipitridae provide a more comprehensive example. 

The files required in your working directory for a PASTIS analysis are:


\begin{table}[h]
\caption{File formats}
    \begin{tabular}{|p{3.9cm} p{1.6cm} p{5.5cm}|}
    \hline
    \textbf{Filename} & \textbf{Required} & \textbf{Description} \\ \hline                                                                                                                                                                                                    
    PatchName.tree & Yes & Constraint tree containing a subset of all the taxa in your patch. This structure will be included in all output trees and should be basedon the best analysis available for those taxa with good information \\
    PatchName.taxa & Yes & List of all the taxa in your tree and their membership of the various clades in the tree                                                                                                                      \\
    PatchName.missingclades & Optional & List of clades that are not represented in the constraint tree and where the may be placed in the tree                                                                                                        \\
    PatchName.sequences & Optional & Sequence data in FASTA format. This is optional, but will be present in most typical analyses!                                                                                                   \\
    PatchName.template & Optional & This is a template file for the mrBayes output file. It outlines options such as the number of iterations, burn in period etc.                                                                                          \\ \hline
    \end{tabular}
\end{table}

\section{File formats}
For a given analysis PASTIS requires up to four files as previously outlined. 
\subsection{.tree}

The .tree file should be in Newick format and contain a phylogenetic tree for a subset of the taxa. This tree represents the information about this patch that has been well resolved by other studies. We refer to this as the constraint tree. PASTIS will create constraints that force the relationships in this tree to be present in all trees sampled by mrBayes. The constraint tree will typically not include all taxa of interest and need not be fully resolved.


\subsection{.taxa}

The .taxa file contains a list of all taxa along with their clade membership. The file contains a header "taxon,clade" followed by one line for each taxon. As suggested by the header the first column is the name of the taxon, followed by a comma and then the name of the clade to which that taxon belongs.

For example, consider the .taxa file from the  example, the first few lines are:\\
\\
 taxon,clade
\\ Tyto\_alba,Outgroup
\\ Cathartes\_aura,Outgroup
\\ Accipiter\_albogularis,Accipiter	
\\ Accipiter\_badius,Accipiter
\\ Accipiter\_bicolor,Accipiter
\\ Accipiter\_brachyurus,Accipiter
\\ Accipiter\_brevipes,Accipiter
\\ Accipiter\_butleri,Accipiter
\\ Accipiter\_castanilius,Accipiter
\\ ...
\\
\\This defines two taxa ("Tyto\_alba" and "Cathartes\_aura") as members of the clade "Outgroup" and lists several taxa belonging to "Accipiter". 

\subsection{.missingclades}

Some clades may be completely missing from the constraint tree (.tree), i.e. not a single member of the clade is present in .tree. In this circumstance the default is that the missing clade may attach anywhere (relative to the constraint tree) without entering into any of the other clades (see the later section on the monophyly\_constraints and paraphyly\_constraints options). 

In many cases further information may be available regarding the placement of missing clades relative to the constraint tree. This information can be specified in the .missingclades file. 

For each .missingclade it is possible to specify sister clades with which a missing clade must group (include constraint) and groups of clades which that clade may not enter (exclude constraint). These constraints are specified one line at a time. The first column contains the name of the missing clade, the second column contains the type of constraint (include or exclude), subsequent columns contain names of the clades which define that constraint. All of these clades should be represented in the constraint tree. PASTIS will provide a warning if this is not the case.

Consider the following hypothetical example .missingclades file where A is a missing clade and the other clades are represented in the constraint tree:
\\
\\A,include,B,C,D,E
\\A,exclude,B,C
\\
\\The include line specifies that A attaches to that section of the tree defined by the MRCA of B, C, D and E. The exclude line then specifies that cannot enter that section of the tree defined by the MRCA of B and C.

\subsection{.sequences}

This contains the aligned sequences for the analysis. This file must be in fasta format.

\subsection{.template}

PASTIS uses a template for the Nexus file. This defines all the mrBayes options that PASTIS cannot deduce from the input files. This includes information such as the number of chains, burn in period, sampling frequency, model priors etc. If the template file does not exist (or NA is specified to pastis\_full), the default template is used. The default template is provided as a starting point and can be examined using:

default\_output\_template()

You will note that there are four unusual terms in here: "<ntax>", "<nchar>", "<sequences>", "<constraints>" and "<outputfile>". These are placeholders that are automatically replaced by PASTIS with the content generated from input files.

To provide your own template file, simply generate a mrBayes input file of the desired type containing the placeholders as illustrated in the default template.


\section{Diagnostics}
After executing mrBayes on the .nexus file produced by PASTIS analyses can be conducted on the mrBayes output as per usual. PASTIS provides one additional function, conch, that can be used to verify where clades and species missing from the constraint tree have been placed relative to the constraint tree. To use it simple do:
\\
\\conch('PatchName.tree','PatchName.nexus.t')
\\
\\(where PatchName.nexus.t was created by the mrBayes execution).
\\
For each taxon, i, not in the constraint tree this will create a file called 'taxonposition\_i.tree'. This tree contains the original constraint tree with the edge lengths equal to the number of sampled trees in which i was descendant from that edge.

The pastis\_simple can be run without sequence data to generate a mrBayes .nexus file that will sample from the prior only. When sampling from the prior the topology is determined by the constraints, not by the sequence data. This is an efficient way to check that constraints have been defined correctly prior to a full analysis.

\section{Constraints}
The PASTIS functions pastis\_simple and pastis\_main contain two parameters that control the extent to which missing clades and taxa can move around the tree. These parameters are monophyly\_constrains and paraphyly\_constrains.

\subsection{monophyly\_constrains}

If set to TRUE (default), missing clades and taxa may not enter into monophyletic groups of taxa. This should only be set to FALSE in unusual circumstances, so make sure you know what you are doing!

\subsection{paraphyly\_constrains}

When set to TRUE (default) missing clades and taxa may not enter into clades that are paraphyletic (i.e. mixed).  This should only be set to FALSE in unusual circumstances, make sure you know what you are doing!


\end{document}
